src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig

```diff
@@ -245,6 +250,11 @@
     {%- endif -%}
 {%- endblock checkbox_radio_label %}
 
+{% block file_label -%}
+    {%- set label_attr = label_attr|merge({class: (label_attr.class|default('') ~ ' custom-file-label')|trim}) -%}
+    {{- block('form_label', 'form_div_layout.html.twig') -}}
```

Comment:
Do you use `form_div_layout.html.twig` to avoid errors in the label?

Reply:
As I wrote on top, we don't need to merge 2 class: "custom-file-label" and "form-control-label".
Or am I wrong?

Reply:
Im not sure. Ill check later tonight

Reply:
In this case is it not better to replace (`replace` filter) an existing class in place of do a double override on the block ? Because if we extend a file, that extend a file, each file override a block, and finally we re-override the block again, it's more difficult to understand what we do, no ?

Reply:
Yes, you're right. But, How to get input type in form_label for write logic with if.
I mean something like that:
```
{% if type == 'file' %} custom-file-label {% else %} form-control-label {% endif %}
```

Reply:
Yes, in the `block form_label`, you can add this:

    {% elseif type is defined and type == 'file' %}
        {%- set label_attr = label_attr|merge({for: id, class: (label_attr.class|default('') ~ ' custom-file-label')|trim}) -%}

To obtain:

    {%- if compound is defined and compound -%}
        {%- set element = 'legend' -%}
        {%- set label_attr = label_attr|merge({class: (label_attr.class|default('') ~ ' col-form-label')|trim}) -%}
    {% elseif type is defined and type == 'file' %}
        {%- set label_attr = label_attr|merge({for: id, class: (label_attr.class|default('') ~ ' custom-file-label')|trim}) -%}
    {%- else -%}
        {%- set label_attr = label_attr|merge({for: id, class: (label_attr.class|default('') ~ ' form-control-label')|trim}) -%}
    {%- endif -%}

Then, you can remove your the block `file_label`. Keep the `type is defined` test, because it appears that it is not always defined.

Reply:
Ha, and then you need to change something fo errors, because by using our `form_label`, you have the errors displayed in it.

I've try, and if you remove the error handling in your `file_row`, error are displayed semi-correctly... But I think it's ok.

In this case, you can both: use your `file_row`, or add a check on `type == file` in the `form_row`. I think it's more readable with your way, but we re-write code, for it, do as you wish :-) But you need to remove the `form_error` from the `file_row` since you use our `form_label` to avoid error duplication.

Reply:
@mpiot Thank you for your help.